### PR TITLE
refactor: use the cli-server-api package

### PR DESCRIPTION
**Pull request by fson**
Summary:
---------

The development server middleware and WebSocket servers were extracted out to the `@react-native-community/cli-server-api` package in #1118. This PR makes the CLI use that package and removes any remaining duplicates of these modules that have been moved.

Test Plan:
----------

Followed the [Testing your changes](https://github.com/react-native-community/cli/blob/66a0616415014c0e8f3da55c0cefffa882718cad/CONTRIBUTING.md#testing-your-changes) instructions in `CONTRIBUTING.md` and started the server with `npx react-native start --watchFolders /Users/ville/Projects/react-native-cli/ --resetCache`. Verified the following features work:
- Running the app on iOS simulator
- Editing a file in the app, the app refreshed automatically
- Refreshing the app by pressing `r`
- Starting the debugger in Chrome and setting a breakpoint
- Looking at logs, inspecting elements and refreshing the app in Flipper

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,13 +33,13 @@ And then:
 
 ```sh
 cd /my/new/react-native/project/
-yarn link "@react-native-community/cli-platform-ios" "@react-native-community/cli-platform-android" "@react-native-community/cli" "@react-native-community/cli-types" "@react-native-community/cli-tools" "@react-native-community/cli-debugger-ui"
+yarn link "@react-native-community/cli-platform-ios" "@react-native-community/cli-platform-android" "@react-native-community/cli" "@react-native-community/cli-server-api" "@react-native-community/cli-types" "@react-native-community/cli-tools" "@react-native-community/cli-debugger-ui"
```
Comment (thymikee) [inline CONTRIBUTING.md:36]: <3

File: packages/cli/src/commands/server/runServer.ts [hunk 2]
```diff
@@ -9,13 +9,13 @@
 import Metro from 'metro';
```
Comment (thymikee) [inline packages/cli/src/commands/server/runServer.ts:9]: Can we also rename the directory from `server` to `start` now that most of the implementation moved to other package? I think it's a good moment to make it happen and be consistent with other commands

Comment (fson) [reply packages/cli/src/commands/server/runServer.ts:9]: Sure, renamed it to `start`.

**Comment by thymikee** [comment]
Noticed an issue with "Reload app" button in Chrome (works fine with Flipper), opening another debugger tab unnecessarily. Same when pressing Reload from an app. Tested on Android

File: packages/cli-server-api/src/index.ts [hunk 3]
```diff
@@ -79,6 +79,7 @@ export function createDevServerMiddleware(options: MiddlewareOptions) {
         '/message',
       );
       broadcast = messageSocket.broadcast;
+      isDebuggerConnected = debuggerProxy.isDebuggerConnected;
```
Comment (thymikee) [inline packages/cli-server-api/src/index.ts:82]: cc @fson the debugger connected state was not updated, likely an oversight when moving code around

**Pull request merged**